articles/getting_to_know/howto/content_pipeline/HowTo_ExtendFontProcessor.md

```diff
@@ -0,0 +1,206 @@
+
+Final code:
+
+```csharp
+using Microsoft.Xna.Framework.Content.Pipeline;
+using Microsoft.Xna.Framework.Content.Pipeline.Graphics;
+using Microsoft.Xna.Framework.Content.Pipeline.Processors;
+using System.ComponentModel;
+using System.IO;
+
+namespace FontProcessor
+{
+    [ContentProcessor(DisplayName = "MyFontProcessor")]
+    internal class MyFontProcessor : FontDescriptionProcessor
+    {
+        public override SpriteFontContent Process(FontDescription input, ContentProcessorContext context)
+        {
+            string fullPath = Path.GetFullPath(MessageFile);
+
+            context.AddDependency(fullPath);
+            string letters = File.ReadAllText(fullPath, System.Text.Encoding.UTF8);
+
+            foreach (char c in letters)
+            {
+                input.Characters.Add(c);
+            }
+            return base.Process(input, context);
+        }
+
+        [DefaultValue("messages.txt")]
+        [DisplayName("Message File")]
+        [Description("The characters in this file will be automatically added to the font.")]
+        public string MessageFile
+        {
+            get { return messageFile; }
+            set { messageFile = value; }
+        }
+        private string messageFile = @"../messages.txt";
+    }
+}
+```
+
+### Associate the custom font processor with the sprite font in the MGCB tool
+
+1. Compile the solution to build **MyFontProcessor**, as you need to add your custom font processor as an available content processor for the content pipeline.
+
+1. Open the MGCB tool for your `Content` project, click (select) the **Content** node, navigate down to the properties section and then click in the **References** property.
+
+    [MGCB Editor window properties](./images/mgcb_editor_content_properties_referencesSelected.png)
+
+1. Navigate to the `dll` for the built `MyFontProcessor` project (which by default is in `\FontGame\MyFontProcessor\bin\debug\net6.0`), select it and click "open".
+
+    > [IMPORTANT]
+    > Ensure that the processor project is always up to date when the main game is built, you need to create a project dependency.  Also use either "debug" or "Release" build types for the processor project, the MGCB tool will NOT dynamically select it when you change the Projects build type.
+
+1. Build the Content Project to ensure everything is connected as it should be.
+
+1. Select the `.spritefont` file, and then in the **Properties** window change the `processor` to`MyFontProcessor` in the drop-down list associated with the **ContentProcessor** field.
+
+When you build the solution, the new processor adds the characters in the messages.txt file to the list of characters available to the [SpriteFont](xref:Microsoft.Xna.Framework.Graphics.SpriteFont).
+
+> [!NOTE]
+> To debug a Content Pipeline importer or processor, add the following line to the processor code to launch the debugger.
+>
+> ```csharp
+> System.Diagnostics.Debugger.Launch();

```

Comment:
This wont work on Mac or Linux (or in VScode). They will need code similar to this https://github.com/MonoGame/MonoGame/pull/8227/files#diff-e4563de56c553b23a664dc5ba2fe0f8f1d1ed1ccb61c275e089c8ea20bf4cd0dR52.